azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -715,6 +724,117 @@ func resourceArmKubernetesClusterDelete(d *schema.ResourceData, meta interface{}
 	return nil
 }
 
+func azureMonitorRoleAssignment(d *schema.ResourceData, meta interface{}) {
+	servicePrincipalProfile := expandAzureRmKubernetesClusterServicePrincipal(d)
+
+	// Get the root roleDefinitionID full path (i.e. providers/Microsoft.Authorization/roleDefinitions/3913510d-42f4-4e42-8a64-420c390055eb)
+	rootRoleDefinitionID := getRootRoleDefinitionID(meta)
+
+	if rootRoleDefinitionID == "" {
+		log.Printf("[INFO] Unable to locate 'Monitoring Metrics Publisher' Root Role Definition Id\n")
+		return
+	}
+
+	// Set the scope to the AKS cluster (i.e. subscriptions/<subscription id>/resourcegroups/<resource group name>/providers/Microsoft.ContainerService/managedClusters/<resource name>)
+	scope := strings.TrimPrefix(d.Id(), "/")
```

Comment:
Jeffrey, scope should be fully qualified resource id that includes "/". what is the reason to trim the "/" prefix?

Reply:
because the REST API adds the leading `/`, if I don't trim the prefix I end up with a scope that starts like `//subscriptions/`

Reply:
> because the REST API adds the leading `/`, if I don't trim the prefix I end up with a scope that starts like `//subscriptions/`

If this is the case, then we are good on this.